chain/vm/syscalls.go

```diff
@@ -88,6 +88,15 @@ func (ss *syscallShim) VerifyConsensusFault(a, b, extra []byte) (*runtime.Consen
 		return nil, xerrors.Errorf("cannot decode second block header: %f", decodeErr)
 	}
 
+	var blockAParent0, blockBParent0 types.BlockHeader
+	if decodeErr := blockAParent0.UnmarshalCBOR(bytes.NewReader(blockA.Parents[0].Bytes())); decodeErr != nil {
```

Comment:
this is just going to try and unmarshal a cid into a block header, which will always fail. We don't have access to both parents here.